website/docs/reference/resource-configs/grants.md

```diff
@@ -1,95 +1,69 @@
 ---
-resource_types: all
-datatype: "{<dictionary>}"
+resource_types: [models,seeds,snapshots]
+datatype: "string"
 default_value: {}
+id: "grants"
 ---
 
+<Snippet src="available-prerelease-beta-banner" />
+
+You can manage access to the datasets you're producing with dbt by using grants. To implement these permissions, define grants as resource configs on each model, seed, or snapshot. Define the default grants that apply to the entire project in your `dbt_project.yml`, and define model-specific grants within each model's SQL or YAML file.
+
+The grant resource configs enable you to apply permissions at build time to a specific set of recipients and model, seed, or snapshot. When your model, seed, or snapshot finishes building, dbt ensures that the grants on its view or table match exactly the grants you have configured.
+
+dbt always finds the most efficient approach when updating grants, which varies based on the adapter you're using, and whether dbt is replacing or updating an object that already exists. You can always check the debug logs for the full set of grant and revoke statements that dbt runs.
```

Comment:
💯 Thanks for including this! I think the Big Idea expressed here is important for users to grok what's expected of them, and what's going on behind the scenes.

I'd ask that we just soften this promise very slightly:

> dbt always finds the most efficient approach when updating grants

"dbt aims to use the most efficient approach"

Just because we know it won't be 100% perfect, there are always edge cases, and we'll always prefer to be safe rather than sorry — meaning, more queries / less efficient to still guarantee the correct set of `grants` are applied

Reply:
Love it! Addressed here: https://github.com/dbt-labs/docs.getdbt.com/pull/1642/commits/b0ca061d74376fab084006a1d8a77d3e1da2d0f7